evm/consensus/pow.py

```diff
@@ -15,11 +24,31 @@
 )
 
 
+cache_seeds = ['\x00' * 32]
+cache_by_seed = OrderedDict()
+cache_by_seed.max_items = 10
```

Comment:
If you want to add `pylru` to the dependencies to get access to a legit LRU cache I've got no problems with that.  Your call.